src/redismodule.h

```diff
@@ -321,6 +321,7 @@ typedef struct RedisModuleCommandArg {
     const char *token;        /* If type is PURE_TOKEN, this is the token. */
     const char *summary;
     const char *since;
+    const char *deprecated_since;
     int flags;                /* The REDISMODULE_CMD_ARG_* macros. */
     struct RedisModuleCommandArg *subargs;
 } RedisModuleCommandArg;
```

Comment:
Doesn't this break modules compiled with a different Redis version and vice versa?

We can only add fields in the end of the struct. The offsets of existing fields are not expected to ever change. See RedisModuleCommandInfoVersion and its handling.

If we want to catch errors like this, we'd need some test involving different versions of Redis.

Reply:
generally yes, but since we didn't release 7.0 yet we can still make changes here

Reply:
well, bumping a version is relatively easy in the mechanism we created, the users (modules) won't even have to be aware of the fact we have two versions...
our code will need to have an additional `if` to check if we can read that var or not.

but anyway, i think that considering this is still an RC, and i suppose that even if some modules started to use this API, it's still in development and nothing was released, so i think in this case we can just let it slide.

Reply:
If we're adding it after the last field (subargs) in the struct we don't risk anything and don't need any additional `if` nor bump the version.

Reply:
IIUC we will need another `if` (if a module that uses this struct compiled with an old redismodule.h, but runs on a new redis-server)

module.c needs to know that it can't read the last 8 bytes of that struct

or maybe I don't fully understand the versioning mechanism here

Reply:
the versioning mechanism had two concerns, one is to know the size of the struct so that arrays can be handled. that's already implicitly taken care of.
but the other is not to try to read values that are not there, for that we need to bump the version, or add an `if` that compares the offset of the field to the size of the struct and concludes if it's there or not.

i think in this case we can just ignore it.

Reply:
OK